src/views/Nft/market/Collection/IndividualNFTPage/PancakeBunnyPage/ForSaleTableCard/index.tsx

```diff
@@ -53,26 +57,61 @@ const ForSaleTableCard: React.FC<ForSaleTableCardProps> = ({ bunnyId, nftMetadat
     setPage,
     direction: priceSort,
     setDirection,
+    isFetchingNfts,
     isLastPage,
     isValidating,
-  } = usePancakeBunnyOnSaleNfts(bunnyId, nftMetadata, itemsPerPage)
+  } = usePancakeBunnyOnSaleNfts(bunnyId, nftMetadata, itemsPerPage * 10)
 
-  const { t } = useTranslation()
-  const { theme } = useTheme()
+  const [internalPage, setInternalPage] = useState(1)
```

Comment:
could you help me understand why we need `internalPage` while we already have `page` from `usePancakeBunnyOnSaleNfts`

Reply:
I assume it would be used to maintain the updated list ( from SC ) and the list ( from Subgraph ). 

Instead of trying the maintain the table state, why don't we maintain the row state only. For example, when we load the list from Subgraph, we check that list status from SC. If items in the Subgraph are no longer tradable, we just disable the `Buy` button. 

In this way, we don't have to maintain another table

Reply:
We need to fetch in batches then chunk it on the view side otherwise if nfts sold or price changed individual pages will show less items then intended. With current behaviour only end of the batch we will show less items to the user.

Reply:
> I assume it would be used to maintain the updated list ( from SC ) and the list ( from Subgraph ).
> 
> Instead of trying the maintain the table state, why don't we maintain the row state only. For example, when we load the list from Subgraph, we check that list status from SC. If items in the Subgraph are no longer tradable, we just disable the `Buy` button.

This would not cover the price changes which user would have to wait delayed subgraph api after making a change in listing price. Thats the other reason we need to fetch in batch then order by price in view side

Reply:
Done